### ISSUE TITLE
Monitor dies when there is no internet connection.

### ISSUE THREAD

**Issue by robertnishihara** [ray-project/ray#1534]
### System information
- **Ray installed from (source or binary)**: source
- **Ray version**: `0.3.1`
- **Exact command to reproduce**:

1. Turn off your internet.

2. Run the following in a Python interpreter.

    ```python
    import ray
    ray.init()
    ```

The following will be printed in the background.

```
Traceback (most recent call last):
  File "/Users/rkn/Workspace/ray/python/ray/monitor.py", line 676, in <module>
    monitor = Monitor(redis_ip_address, redis_port, autoscaling_config)
  File "/Users/rkn/Workspace/ray/python/ray/monitor.py", line 99, in __init__
    self.load_metrics = LoadMetrics()
  File "/Users/rkn/Workspace/ray/python/ray/autoscaler/autoscaler.py", line 101, in __init__
    self.local_ip = services.get_node_ip_address()
  File "/Users/rkn/Workspace/ray/python/ray/services.py", line 203, in get_node_ip_address
    s.connect((ip_address, int(port)))
OSError: [Errno 51] Network is unreachable
```

This is related to #351.

@ericl do you think that `ray.services.get_node_ip_address()` should just return `127.0.0.1` if there is no internet connection?

**Comment by ericl** [comment]
Seems reasonable, the node probably isn't part of a cluster if the network is down.

**Comment by robertnishihara** [comment]
It could be part of a cluster that is not connected to the internet. This just means that it can't connect to google I think.

**Comment by ericl** [comment]
Ic, in that case the head node needs to know it's private ip to get workers connected to it. So 127.0.0.1 won't work. Is it possible to resolve the private ip even without internet access?

**Comment by robertnishihara** [comment]
Fixed by #1556.
